modules/worker_groups/data.tf

```diff
@@ -0,0 +1,35 @@
+data "aws_caller_identity" "current" {}
+data "aws_partition" "current" {}
+
+data "aws_iam_instance_profile" "custom_worker_group_iam_instance_profile" {
+  for_each = var.manage_worker_iam_resources ? {} : local.worker_group_configurations
+
+  name = each.value["iam_instance_profile_name"]
+}
+
+data "aws_ami" "eks_worker" {
+  filter {
```

Comment:
can we use for_each and add support for worker_node_version specified in the var.workers_group and the default value will be cluster_version if worker_node_version not specified

Reply:
What do you mean by 'worker_node_version'?